scheduler/pkg/scheduler/filters/serverrequirements_test.go

```diff
@@ -39,28 +39,40 @@ func TestReplicaRequirementsFilter(t *testing.T) {
 			store.ModelProgressing)
 	}
 
-	getTestServerReplicaWithCaps := func(capabilities []string) *store.ServerReplica {
+	makeServerReplica := func(server *store.Server, capabilities []string) *store.ServerReplica {
 		return store.NewServerReplica("svc", 8080, 5001, 1, store.NewServer("server", true), capabilities, 100, 100, 0, nil, 100)
 	}
 
+	makeServer := func(replicas int, capabilities []string) *store.ServerSnapshot {
+		server := store.NewServer("server", true)
+		snapshot := server.CreateSnapshot(false, false)
+
+		for i := 0; i < replicas; i++ {
+			replica := makeServerReplica(server, capabilities)
+			snapshot.Replicas[i] = replica
+		}
+
+		return snapshot
+	}
+
 	type test struct {
 		name     string
 		model    *store.ModelVersion
-		server   *store.ServerReplica
+		server   *store.ServerSnapshot
 		expected bool
 	}
 
 	tests := []test{
```

Comment:
:wrench: There should be new tests here to cover the case of zero replicas and maybe multiple

Reply:
Added in https://github.com/SeldonIO/seldon-core/pull/5107/commits/a527b600978b77d0322d4bea795de3562becd4ce and https://github.com/SeldonIO/seldon-core/pull/5107/commits/bb22ca22bf1e75da01469770c3c61fe9b9e7da97